include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -518,8 +518,14 @@ struct __usm_or_buffer_accessor
     }
 };
 
+struct __result_and_scratch_storage_base
+{
+    virtual ~__result_and_scratch_storage_base() = default;
+};
+using __result_and_scratch_storage_base_ptr = std::shared_ptr<__result_and_scratch_storage_base>;
```

Comment:
I am inclined to remove this `using` line, to not "hide" the fact that this is a **shared** pointer.  I think this adds to some of the confusion surrounding this mechanism.  
At the very least, we should use "shared_ptr" in the name, but at that point, the name would just as long as the more descriptive original type expression.

Reply:
Fixed.